### PR TITLE
add_user_functionality_and_db

### PR DESCRIPTION
**Pull request by DaniilPoiarkov**

### CODE REVIEW

**Review by AndriiMatchak — CHANGES_REQUESTED** [review]

File: PersonalPreject.DataAccess/AppDbContext.cs [hunk 1]
```diff
@@ -14,5 +15,12 @@ public AppDbContext(DbContextOptions<AppDbContext> options)
         {
 
         }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+
+        }
+
+        public DbSet<User> Users { get; set; }
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalPreject.DataAccess/AppDbContext.cs:24]: Resolved

File: PersonalProject.Common/Models/IncomeModels/CreateUserIncomeModel.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        //public string? FirstName { get; set; }
+
+        //public string? LastName { get; set; }
+
+        //public int? Age { get; set; }
+
+        //public Sex? Sex { get; set; }
+
+        //public bool? IsEmailConfirmed;
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject.Common/Models/IncomeModels/CreateUserIncomeModel.cs:24]: Resolved

File: PersonalProject.Common/Models/IncomeModels/UpdateUserIncomeModel.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        public string FirstName { get; set; }
+
+        public string LastName { get; set; }
+
+        public int Age { get; set; }
+
+        public Sex Sex { get; set; }
+
+        public bool IsEmailConfirmed { get; set; }
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject.Common/Models/IncomeModels/UpdateUserIncomeModel.cs:21]: Resolved

File: PersonalProject.Services/Implementation/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                UpdatedDate = user.UpdatedDate,
+                IsEmailConfirmed = user.IsEmailConfirmed,
+            };
+        }
+
+        public void Update(Guid id, UpdateUserIncomeModel model)
+        {
+            var user = _userRepository.Get(u => u.Id == id);
+
+
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject.Services/Implementation/UserService.cs:64]: Resolved

File: PersonalProject.Services/Implementation/UserService.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            {
+                Id = u.Id,
+                Email = u.Email,
+                Password = u.Password,
+                FirstName = u.FirstName,
+                LastName = u.LastName,
+
+            }).ToList();
+        }
+
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject.Services/Implementation/UserService.cs:85]: Resolved

File: PersonalProject/Controllers/UserController.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using PersonalProject.Common.Models.BaseModel;
+using PersonalProject.Common.Models.EntityDTOs;
+using PersonalProject.Common.Models.IncomeModels;
+using PersonalProject.Services.Interfaces;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject/Controllers/UserController.cs:7]: Resolved

File: PersonalProject/Program.cs [hunk 7]
```diff
@@ -1,9 +1,25 @@
...
+string connection = builder.Configuration.GetConnectionString("DefaultConnection");
+
+builder.Services.AddDbContext<AppDbContext>(options => options.UseLazyLoadingProxies().UseSqlServer(connection));
+
 builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
+builder.Services.AddScoped<IUserService, UserService>();
+builder.Services.AddTransient(typeof(IBaseRepository<>), typeof(BaseRepository<>));
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject/Program.cs:19]: Resolved

File: PersonalProject/Controllers/UserController.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace PersonalProject.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [ApiController]
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject/Controllers/UserController.cs:12]: Resolved

File: PersonalProject.Services/Implementation/UserService.cs [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    public class UserService : IUserService
+    {
+        private readonly IBaseRepository<User> _userRepository;
+
+        public UserService(IBaseRepository<User> userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        public void Create(CreateUserIncomeModel model)
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject.Services/Implementation/UserService.cs:26]: Resolved

File: PersonalProject/Program.cs [hunk 10]
```diff
@@ -1,9 +1,25 @@
...
+using PersonalProject.Repository;
+using PersonalProject.Repository.Interfaces;
+using PersonalProject.Services.Implementation;
+using PersonalProject.Services.Interfaces;
+
 var builder = WebApplication.CreateBuilder(args);
 
+string connection = builder.Configuration.GetConnectionString("DefaultConnection");
+
+builder.Services.AddDbContext<AppDbContext>(options => options.UseLazyLoadingProxies().UseSqlServer(connection));
```
[the remark this answers is not in the document]
Comment (DaniilPoiarkov) [reply PersonalProject/Program.cs:12]: Resolved

**Review by AndriiMatchak — APPROVED** [review]

**Pull request merged**